### PR TITLE
Address send token objective feedback

### LINKED ISSUE #4450: Token usage objective review feedback
### Description

After the token usage objective view, the team gave the following reviews, clarifications and improvement suggestions;

#### Send tokens screen

- Should the fromApplication not be only the current application instead of a list?
   - What applications are in the dropdown list? User’s added applications?
- What are the default fromApplication and toApplication? Is it the current application?
- Consider adding a space between value and unit in “Balance: 0EVT”
- Amount in fiat (show only for LSK)
- Priority: Display the priority fees as in mobile - for consistency reasons?
- Initiation fee: Consider renaming it to `initialization` fee instead for consistency with the terminology
- Show total fees as well as a sum of all the components to make it easier for the user
- Is service to provide the CCM/initiation fees?? — New requirements? Or, is this information coming from the SDK?
- Transaction failure messages are different on mobile and desktop wallets — sync and keep them consistent
- Needs edge-case scenario handling: when both the fromApplication and toApplication support all the tokens.
- Chars left while typing message
- Application and native token of that chain should be preselected
- Dropdown border colour not matching design
- `Send LSK` button should be generic to “Send Token”
- Add token logos (always showing LISK)
- Editing transaction fee on custom priority crashes the application
- Tooltip on the priority  section of the summary page isn’t correct

#### Request tokens screen

- Token field should be enabled only after the recipient application is chosen, must be disabled by default
- How is the recipient address determined here? Based on the logged in user, I believe?

### PR DESCRIPTION
**Pull request by eniolam1000752**
### What was the problem?

This PR resolves #4450

### How was it solved?

- [x] Introduce feedback to the `MessageField` component
- [x] Modify AmountField to validate against maximum decimal value
- [x] fix failing unit tests

### How was it tested?

- Jest
- Visually

### CODE REVIEW

File: src/const/regex.js [hunk 1]
```diff
@@ -23,13 +25,19 @@ export const regex = {
   amount: {
     en: {
       format: /[^\d.]|(.*?\.){2}|\.$/,
-      maxFloating: /\.[\d\w\s]{9}/,
       leadingPoint: /^\./,
+      maxFloating: (token) => {
```
Comment (ManuGowda) [inline src/const/regex.js]:
```suggestion
      maxDecimals: (token) => {
```

File: src/locales/en/common.json [hunk 2]
```diff
@@ -335,7 +335,7 @@
   "Max amount must be greater than Min amount": "Max amount must be greater than Min amount",
   "Max. 20 characters": "Max. 20 characters",
   "Max. 20 characters, a-z, 0-1, no special characters except !@$_.": "Max. 20 characters, a-z, 0-1, no special characters except !@$_.",
-  "Maximum floating point is 8.": "Maximum floating point is 8.",
+  "Maximum allowed decimal point is 8.": "Maximum allowed decimal point is 8.",
```
Comment (ManuGowda) [inline src/locales/en/common.json]: Shouldn't this be template? based on your implementation 

```
message: i18n.t(`Maximum allowed decimal point is ${getTokenDecimals(token)}.`),
```

File: src/modules/token/fungible/components/MessageField/MessageField.js [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 import { TertiaryButton } from 'src/theme/buttons';
+import Feedback from 'src/theme/feedback/feedback';
 import Icon from 'src/theme/Icon';
 import CircularProgress from 'src/theme/ProgressCircular/circularProgress';
 import Tooltip from 'src/theme/Tooltip';
 import { sizeOfString } from 'src/utils/helpers';
 import styles from './MessageField.css';
 
+const messageErrorFeedback = {
+  maxCharSize: 'Message size has exceeded the maxium allowed size',
```
Comment (ManuGowda) [inline src/modules/token/fungible/components/MessageField/MessageField.js]:
```suggestion
  maxCharSize: 'Message size has exceeded the maximum allowed size',
```

File: src/modules/token/fungible/components/MessageField/MessageField.js [hunk 4]
```diff
@@ -23,12 +28,17 @@ function MessageField({
...
   const byteCount = useMemo(() => sizeOfString(value), [value]);
+  const [feedBackType, setFeedBackType] = useState(null);
 
   const onShrinkField = useCallback(() => {
     setCollapsed(!isCollapsed);
     onRemove?.();
   }, [isCollapsed]);
 
+  useEffect(() => {
+    setFeedBackType(maxMessageLength - byteCount < 0 ? 'maxCharSize' : null);
```
Comment (ManuGowda) [inline src/modules/token/fungible/components/MessageField/MessageField.js]:
```suggestion
    setFeedbackType(maxMessageLength - byteCount < 0 ? 'maxCharSize' : null);
```

File: src/modules/token/fungible/hooks/useAmountField.js [hunk 5]
```diff
@@ -64,23 +64,36 @@ const useAmountField = (initialValue, balance, token) => {
   }, [initialValue]);
 
   const onAmountInputChange = ({ value }, maxAmount) => {
-    const { leadingPoint } = regex.amount[i18n.language];
+    const { leadingPoint, maxFloating } = regex.amount[i18n.language];
     value = leadingPoint.test(value) ? `0${value}` : value;
-    clearTimeout(loaderTimeout);
+    const isMaxAccuracyInValid = maxFloating(token).test(value);
```
Comment (ManuGowda) [inline src/modules/token/fungible/hooks/useAmountField.js]:
```suggestion
    const isAmountValid = maxFloating(token).test(value);
```

File: src/modules/token/fungible/hooks/useAmountField.js [hunk 6]
```diff
@@ -64,23 +64,36 @@ const useAmountField = (initialValue, balance, token) => {
   }, [initialValue]);
 
   const onAmountInputChange = ({ value }, maxAmount) => {
```
Comment (ManuGowda) [inline src/modules/token/fungible/hooks/useAmountField.js]: can we use `amount`? that way its more readable, i.e `({ value: amount }, maxAmount)`

**Review by ManuGowda — CHANGES_REQUESTED** [review]

**Review by ManuGowda — DISMISSED** [review]

**Review by ManuGowda — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**